src/avatar.h

```diff
@@ -188,6 +203,12 @@ class avatar : public Character
 
         void remove_active_mission( mission &cur_mission );
 
+        /**
+        * Despite the name, this operation also makes an existing point of interest active.
```

Comment:
You could separate that out into its own function. So your regular function call would look like

`activate_point_of_interest( add_point_of_interest ( foo ) )`

Where `point_of_interest()` would return a copy or reference to the addition. (If you also did the std::optional change you could have it return std::nullopt in case of failure. Though there shouldn't be a chance for it to fail here.)